### PR TITLE
Update docs to point to discussions, wiki, etc

### LINKED ISSUE #2471: Grids documentation is out of date
The documentation on our grids is quite out of date:

https://clima.github.io/OceananigansDocumentation/stable/model_setup/grids/

It claims that we only support `RectilinearGrid`, and doesn't mention how to build a grid on the GPU.

### PR DESCRIPTION
**Pull request by glwagner**
This PR updates the docs to provide stronger links to the [new wiki](https://github.com/CliMA/Oceananigans.jl/wiki).

There's a few more updates I'd like to make, and perhaps we can completely migrate "using GPUs" and "installation instructions" over to the new wiki? @tomchor and @navidcy are the docs aficionados so let's see what they think.

~I'd also like to resolve #2471 .~

Closes #2471

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -2,28 +2,51 @@
...
+Oceananigans is a [registered Julia package](https://julialang.org/packages/), so the installation instructions are:
+
+1. [Download Julia 1.6](https://julialang.org/downloads/);
+
+2. Start up Julia and type
+
+```julia
+julia> using Pkg
+
+julia> Pkg.add("Oceananigans")
```
Comment (navidcy) [inline docs/src/index.md:24]:
```suggestion
julia> using Pkg; Pkg.add("Oceananigans")
```

**Review by navidcy — APPROVED** [review]

File: docs/src/index.md (hunk 1 above)
Comment (navidcy) [reply docs/src/index.md:24]: just a suggestion -- either way really!

Comment (glwagner) [reply docs/src/index.md:24]: I was thinking the two line version is easier to read, but the one line version is nice and snappy

File: docs/src/index.md [hunk 2]
```diff
@@ -2,28 +2,51 @@
...
-## Getting help
-
-If you are interested in using Oceananigans.jl or are trying to figure out how to use it, please feel free to ask us
-questions and get in touch! If you're trying to set up a model then check out the examples and model setup
-documentation. Please feel free to [start a discussion](https://github.com/CliMA/Oceananigans.jl/discussions)
-if you have any questions, comments, suggestions, etc! There is also an #oceananigans channel on the
-[Julia Slack](https://julialang.org/slack/).
+Oceananigans is fast, friendly, and flexible software for finite volume simulations the nonhydrostatic
+and hydrostatic Boussinesq equations on CPUs and GPUs. Aside from its ability to run on GPUs (wow, fast!)
+Oceananigans receives it's powers from a flexible and intuitive user interface that
```
Comment (navidcy) [inline docs/src/index.md:7]:
```suggestion
Oceananigans receives its powers from a flexible and intuitive user interface that
```

File: docs/src/index.md (hunk 1 above)
Comment (navidcy) [reply docs/src/index.md:24]: leave it with 2 lines!

**Comment by glwagner** [comment]
I moved much of the installation instructions into

https://github.com/CliMA/Oceananigans.jl/wiki/Productive-Oceananigans-workflows-and-Julia-environments

and a little bit into

https://github.com/CliMA/Oceananigans.jl/wiki/Installation-and-getting-started-with-Oceananigans

**Comment by glwagner** [comment]
I think we should rename "simulation tips" to something like "caveats". All those things aren't really specific to simulations anyways, and they mostly regard "gotchas" and things that can go wrong. It's very useful to have documentation on that so let's label it right!

File: docs/src/quick_start.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Quick start
+
+This code:
+
+```jldoctest cpu
```
Comment (navidcy) [inline docs/src/quick_start.md:5]: why are these doctests? was this on purpose?

Comment (glwagner) [reply docs/src/quick_start.md:5]: hmm no. I think I want to literate this thing.

Comment (navidcy) [reply docs/src/quick_start.md:5]:
```@example

?

File: docs/src/quick_start.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+using GLMakie
+
+u, v, w = model.velocities
+ζ = Field(∂x(v) - ∂y(u))
+compute!(ζ)
+
+heatmap(interior(ζ, :, :, 1))
+```
+
+A few more time-steps, and it's starting to get a little diffuse!
```
Comment (navidcy) [inline docs/src/quick_start.md:35]:
```suggestion
A few more time-steps, and it's starting to get a little diffused!
```

File: docs/src/quick_start.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+This code:
+
+```@example cpu
+using Oceananigans
+
+grid = RectilinearGrid(size=(128, 128), halo=(3, 3), x=(0, 2π), y=(0, 2π), topology=(Periodic, Periodic, Flat))
+model = NonhydrostaticModel(; grid, advection=WENO5())
+
+ϵ(x, y, z) = 2rand() - 1
```
Comment (navidcy) [inline docs/src/quick_start.md:11]: why this? why not simply `ϵ(x, y, z) = rand()`?

File: docs/src/quick_start.md (hunk 4 above)
Comment (glwagner) [reply docs/src/quick_start.md:35]: I like diffuse! :-D

**Comment by glwagner** [comment]
Preview here: https://clima.github.io/OceananigansDocumentation/previews/PR2474/

File: README.md [hunk 6]
```diff
@@ -122,7 +122,7 @@ L = Lx = Ly = Lz = 2000  # Length of each dimension.
 topology = (Periodic, Periodic, Bounded)
 
 model = NonhydrostaticModel(
-            grid = RectilinearGrid(CPU(); topology=topology, size=(Nx, Ny, Nz), extent=(Lx, Ly, Lz)),
+            grid = RectilinearGrid(CPU(); topology, size=(Nx, Ny, Nz), extent=(Lx, Ly, Lz)),
```
Comment (glwagner) [inline README.md:125]: I'm wondering if we should use a better example here, what do you think?

Comment (navidcy) [reply README.md:125]: I think it's all right... What would be simpler?

Comment (glwagner) [reply README.md]: I put in a two-dimensional turbulence example that isn't huge and actually does something instead.

**Comment by glwagner** [comment]
I think this is probably ready to merge. The README will need more work soon because we need to update the movies pretty badly with more recent examples (and some sphere stuff too!) So that should be coming relatively soon anyways.

**Comment by navidcy** [comment]
I added a few things on grids. Have a look and edit at your liking.

Yes, merge!

**Pull request merged**